### PR TITLE
I1241 redesign header

### PR DESCRIPTION
**Pull request by Serhiy91**

### CODE REVIEW

File: app/imports/client/react/main-header/components/OrganizationMenu.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import { OrganizationSettingsHelp } from '../../../../api/help-messages';
+import { OrgCurrencies, UserRoles } from '../../../../share/constants';
+import { Query as Queries } from '../../../graphql';
+import { FlowRouterContext, RenderSwitch } from '../../components';
+import { WithToggle } from '../../helpers';
+import HeaderMenuItem from './HeaderMenuItem';
+
+const ViewItemLabels = {
+  canvas: 'Business design view',
+  dashboardPage: 'Business operations view',
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/OrganizationMenu.js]: Rename this.
Also add there `CANVAS: 'canvas'` and `DASHBOARD_PAGE: 'dashboardPage'` and reuse below for any string.

File: app/imports/client/react/main-header/components/OrganizationMenu.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                          />
+                          <HeaderMenuItem
+                            label={ViewItemLabels.dashboardPage}
+                            tag="a"
+                            href={router.path('dashboardPage', { orgSerialNumber })}
+                            active={routeName === 'dashboardPage'}
+                          />
+
+                          <DropdownItem divider />
+                          {me.roles.includes(UserRoles.CHANGE_ORG_SETTINGS) && (
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/OrganizationMenu.js]: What if `me` or `me.roles` is undefined?

File: app/imports/client/react/main-header/components/OrganizationMenu.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                                  _title: 'Organization settings',
+                                  helpText: OrganizationSettingsHelp.organizationSettings,
+                                  organizationId: currentOrg._id,
+                                });
+                              }}
+                            />
+                          )}
+
+                          <HeaderMenuItem
+                            label="Create new Plio organisation"
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/OrganizationMenu.js]: Everywhere else is "organization" and only there "organisation"? We use American english.

File: app/imports/client/react/main-header/components/OrganizationMenu.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                                variation: 'save',
+                                timezone: moment.tz.guess(),
+                                ownerName: me.profile.fullName,
+                                currency: OrgCurrencies.GBP,
+                              });
+                            }}
+                          />
+
+                          <DropdownItem divider />
+                          <HeaderMenuItem label="Plio version 0.3.40" disabled />
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/OrganizationMenu.js]: It's better to define a constant for it in `api/constants`. Assign only version number to it.

File: app/imports/client/react/main-header/index.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export { default as MainHeader } from './components/MainHeader';
```
Comment (Dartv) [inline app/imports/client/react/main-header/index.js]: No need for exports there. If we export something as `{ MainHeader } from '...'` tree shaking suddenly doesn't work, so we should use traditional imports in future for module components. It's fine to import reusable components that way though.

File: app/imports/startup/client/mixins/roles.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ export default {
         organizationId,
       );
     }
+    return null;
```
Comment (Dartv) [inline app/imports/startup/client/mixins/roles.js]: Why not `false`? The name suggests it should be boolean

File: app/imports/ui/components/dashboard/includes/organization-menu/create/create.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import { Template } from 'meteor/templating';
 import { FlowRouter } from 'meteor/kadira:flow-router';
 
-import { Organizations } from '/imports/share/collections/organizations.js';
-import { insert } from '/imports/api/organizations/methods.js';
+import { insert } from '../../../../../../api/organizations/methods.js';
+import '../settings';
```
Comment (Dartv) [inline app/imports/ui/components/dashboard/includes/organization-menu/create/create.js]: try using dynamic import in the event where you open menu.

File: app/imports/ui/components/dashboard/includes/organization-menu/item/item.js [hunk 8]
```diff
@@ -1,22 +0,0 @@
...
-    return `^\\/${this.serialNumber()}`;
-  },
-  selectOrg(e) {
-    e.preventDefault();
-
-    const selectedOrganizationSerialNumber = this.serialNumber();
-
-    this.dispatch(organizationChanged);
-
-    localStorage.setItem(`${Meteor.userId()}: selectedOrganizationSerialNumber`, selectedOrganizationSerialNumber);
```
Comment (Dartv) [inline app/imports/ui/components/dashboard/includes/organization-menu/item/item.js]: You deleted this and didn't reimplement it.

File: app/imports/ui/components/dashboard/includes/organization-menu/menu.html [hunk 9]
```diff
@@ -1,35 +0,0 @@
...
-        {{/each}}
-        {{#if canEditOrgSettings organizationId}}
-          <div class="dropdown-divider"></div>
-          <a class="dropdown-item org-settings-modal-link"
-            {{b "click: openOrgSettings"}}>
-            Organization settings
-          </a>
-        {{/if}}
-
-        {{#if haveCustomerAccess}}
```
Comment (Dartv) [inline app/imports/ui/components/dashboard/includes/organization-menu/menu.html]: I don't see this option in a new dropdown.

File: app/imports/client/react/main-header/components/HeaderMenuItem.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      position: relative;
+      top: 5px;
+      color: ${Styles.color.blue}
+    }
+  }
+`;
+
+const HeaderMenuItem = ({ label, active, ...props }) => (
+  <DropdownItemStyled {...props}>
+    <span>{label}</span>
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/HeaderMenuItem.js]: Why not just use `children` there? Seems more natural

File: app/imports/client/react/main-header/components/OrganizationMenu.js [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+                              onClick={() => router.setParams({ orgSerialNumber: serialNumber })}
+                              active={currentOrg._id === _id}
+                            />
+                          ))}
+
+                          <DropdownItem divider />
+                          <HeaderMenuItem
+                            label={ViewItemLabels.canvas}
+                            tag="a"
+                            href={router.path('canvas', { orgSerialNumber })}
```
Comment (Dartv) [inline app/imports/client/react/main-header/components/OrganizationMenu.js]: Define a new constant `RouteNames` in `api/constants` and reuse it

**Comment by Dartv** [comment]
"Send test notification" in my preferences doesn't work

**Comment by Serhiy91** [comment]
@Dartv 1 and 5 points I cannot reproduce https://drive.google.com/file/d/1dP4quMNI-p38UxxT7LkqDNn_WPg942FA/view. Can you try to retest?

**Comment by Dartv** [comment]
1. My notifications were turned off.
2. Try adding and deleting an org from business operations view

**Comment by Serhiy91** [comment]
https://github.com/Pliohub/plio-util/pull/28 - plio-util

File: app/imports/ui/components/includes/quill-editor/quill-editor.js [hunk 12]
```diff
@@ -6,21 +6,24 @@ Template.QuillEditor.viewmodel({
   isExpanded: false,
   async onRendered(tpl) {
     const { default: Quill } = await import('quill');
-    this.editor(new Quill(this.templateInstance.find('.editor-container')));
-    this.editor().addModule('toolbar', {
-      container: this.templateInstance.find('.editor-toolbar'),
-    });
-    this.editor().addModule('link-tooltip', true);
+    const containerElement = this.templateInstance.find('.editor-container');
```
[the remark this answers is not in the document]
Comment (Serhiy91) [reply app/imports/ui/components/includes/quill-editor/quill-editor.js]: Otherwise, it throws an error.

**Comment by Serhiy91** [comment]
@Dartv fixed

**Comment by Serhiy91** [comment]
@Dartv fixed

**Pull request merged**